<launch>
  <arg name="mav_name" value="firefly"/>
  <arg name="enable_logging" default="false" />
  <arg name="enable_ground_truth" default="true" />
  <arg name="log_file" default="$(arg mav_name)" />
  <arg name="paused" value="true"/>

  <arg name="world_frame_name" default="world"/>
  <arg name="world_resolution" value="0.15"/>
  <arg name="world_name" value="forest0"/>

  <arg name="start_x" value="-15.0"/>
  <arg name="start_y" value="15.0"/>
  <arg name="start_z" value="0.6"/>
  <arg name="start_yaw" value="$(eval -pi/4)"/>

  <arg name="goal1_x" value="0.0"/>
  <arg name="goal1_y" value="0.0"/>
  <arg name="goal1_z" value="0.6"/>
  <arg name="goal1_yaw" value="$(eval -pi/4)"/>

  <arg name="goal2_x" value="5.0"/>
  <arg name="goal2_y" value="-15.0"/>
  <arg name="goal2_z" value="0.6"/>
  <arg name="goal2_yaw" value="$(eval -pi/4)"/>
  
  <arg name="time_limit" value="90" />
  <arg name="dt" value="0.05" />

  <arg name="dummy_max_lat_velo" value="30" />
  <arg name="dummy_max_yaw_velo" value="30" />
  
  <arg name="tyaw_cnt" value="31" />
  <arg name="tpitch_cnt" value="21" />
  <arg name="tsamp_cnt" value="25" />
  <arg name="tlen" value="10" />
  <arg name="tyaw" value="0.942" />
  <arg name="tpitch" value="0.628" />
  <arg name="tentacle_type" value="linear" />
  <arg name="pdist" value="0.35" />
  <arg name="sdist" value="1.5" />
  <arg name="egrid_vdim" value="0.1" />
  <arg name="egrid_vnumx" value="200" />
  <arg name="egrid_vnumy" value="200" />
  <arg name="egrid_vnumz" value="200" />

  <arg name="crash_dist" value="2" />
  <arg name="clear_scale" value="0" />
  <arg name="clutter_scale" value="20" />
  <arg name="close_scale" value="60" />
  <arg name="smooth_scale" value="2" />

  <env name="GAZEBO_MODEL_PATH" value="$(find rotors_gazebo)/models"/>
  <env name="GAZEBO_RESOURCE_PATH" value="$(find forest_gen)"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find forest_gen)/worlds/$(arg world_name).world" />
    <!-- <arg name="debug" value="true"/> -->
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" default="true"/>
    <arg name="verbose" default="false"/>
  </include>

  <group ns="$(arg mav_name)">
    <include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
      <arg name="mav_name" value="$(arg mav_name)" />
      <arg name="model" value="$(find ewok_simulation)/urdf/mav_r200.gazebo" />
      <arg name="enable_logging" value="$(arg enable_logging)" />
      <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
      <arg name="log_file" value="$(arg log_file)"/>
      <arg name="x" value="$(arg start_x)"/>
      <arg name="y" value="$(arg start_y)"/>
    </include>

    <node name="lee_position_controller_node" pkg="rotors_control" type="lee_position_controller_node" output="screen">
      <rosparam command="load" file="$(find rotors_gazebo)/resource/lee_controller_$(arg mav_name).yaml" />
      <rosparam command="load" file="$(find rotors_gazebo)/resource/$(arg mav_name).yaml" />
      <param name="dt" value="$(arg dt)" />
      <remap from="odometry" to="odometry_sensor1/odometry" />
    </node>

    <node name="tentabot_go_extend" pkg="tentabot" type="tentabot_go_extend" output="screen">

      <param name="world_frame_name" value="$(arg world_frame_name)" />
      <param name="world_resolution" value="$(arg world_resolution)" />
      <param name="world_name" value="$(arg world_name)"/>
      <param name="start_x" value="$(arg start_x)" />
      <param name="start_y" value="$(arg start_y)" />
      <param name="start_z" value="$(arg start_z)" />
      <param name="start_yaw" value="$(arg start_yaw)" />
      <param name="goal1_x" value="$(arg goal1_x)" />
      <param name="goal1_y" value="$(arg goal1_y)" />
      <param name="goal1_z" value="$(arg goal1_z)" />
      <param name="goal1_yaw" value="$(arg goal1_yaw)" />
      <param name="goal2_x" value="$(arg goal2_x)" />
      <param name="goal2_y" value="$(arg goal2_y)" />
      <param name="goal2_z" value="$(arg goal2_z)" />
      <param name="goal2_yaw" value="$(arg goal2_yaw)" />
      <param name="time_limit" value="$(arg time_limit)" />
      <param name="dt" value="$(arg dt)" />
      <param name="dummy_max_lat_velo" value="$(arg dummy_max_lat_velo)" />
      <param name="dummy_max_yaw_velo" value="$(arg dummy_max_yaw_velo)" />
      <param name="tyaw_cnt" value="$(arg tyaw_cnt)" />
      <param name="tpitch_cnt" value="$(arg tpitch_cnt)" />
      <param name="tsamp_cnt" value="$(arg tsamp_cnt)" />
      <param name="tlen" value="$(arg tlen)" />
      <param name="tyaw" value="$(arg tyaw)" />
      <param name="tpitch" value="$(arg tpitch)" />
      <param name="tentacle_type" value="$(arg tentacle_type)" />
      <param name="pdist" value="$(arg pdist)" />
      <param name="sdist" value="$(arg sdist)" />
      <param name="egrid_vdim" value="$(arg egrid_vdim)" />
      <param name="egrid_vnumx" value="$(arg egrid_vnumx)" />
      <param name="egrid_vnumy" value="$(arg egrid_vnumy)" />
      <param name="egrid_vnumz" value="$(arg egrid_vnumz)" />
      <param name="crash_dist" value="$(arg crash_dist)" />
      <param name="clear_scale" value="$(arg clear_scale)" />
      <param name="clutter_scale" value="$(arg clutter_scale)" />
      <param name="close_scale" value="$(arg close_scale)" />
      <param name="smooth_scale" value="$(arg smooth_scale)" />
    </node>

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  </group>
</launch>
